dsl_parser/tests/test_blueprint_diff.py

```diff
@@ -0,0 +1,145 @@
+########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import unittest
+
+import yaml
+from mock import patch
+
+
+from dsl_parser.blueprint_diff import convert_chain_to_dict
+from dsl_parser.blueprint_diff import add_chain
+from dsl_parser.blueprint_diff import has_chain
+from dsl_parser.blueprint_diff import get_value
+from dsl_parser.blueprint_diff import blueprint_diff
+from dsl_parser.blueprint_diff import list_diff
+
+THIS_DIR = os.path.dirname(os.path.dirname(__file__))
+TESTS_DIR = os.path.join(THIS_DIR, 'tests')
+RESOURCES_DIR = os.path.join(TESTS_DIR, 'resources')
+
+stub_chain = ['a', 'b', 'c', 'd']
+stub_value = 'value'
+stub_dictionary = {'a': '1'}
+
+
+class DiffTestCase(unittest.TestCase):
+
+    def test_convert_dict_to_chain(self):
+        empty_chain = []
+        chain_with_one_key = ['a']
+        long_chain = ['a', 'b', 'c']
+
+        value = 'added'
+
+        empty_dict = {}
+        dict_with_one_key = {'a': value}
+        nested_dict = {'a': {'b': {'c': value}}}
+
+        self.assertEqual(convert_chain_to_dict(empty_chain, value),
+                         empty_dict
+                         )
+        self.assertEqual(convert_chain_to_dict(chain_with_one_key, value),
+                         dict_with_one_key
+                         )
+        self.assertEqual(convert_chain_to_dict(long_chain, value),
+                         nested_dict)
+
+    @patch('dsl_parser.blueprint_diff.convert_chain_to_dict')
+    def test_add_chain_to_empty_output(self, convert_chain_mock):
+
+        output = {}
+
+        add_chain(output, stub_chain, stub_value)
+
+        convert_chain_mock.assert_called_with(stub_chain[0:], stub_value)
+
+    @patch('dsl_parser.blueprint_diff.convert_chain_to_dict')
+    def test_add_chain_to_output(self, convert_chain_mock):
+
+        output = {'a': {'b': {'c': 3}, 'd': {'e': 5}}, 'f': 6}
+        chain = ['a', 'b', 'h']
+        add_chain(output, chain, stub_value)
+
+        convert_chain_mock.assert_called_with(chain[2:], stub_value)
+
+    def test_has_chain(self):
+
+        output = {'a': {'b': {'c': 3}, 'd': {'e': 5}}, 'f': 6}
+
+        chain_a = ['a', 'b', 'd']
```

Comment:
change that names to more meaningful ones.
like:
non existent
exists
deep exists